code/__HELPERS/type_processing.dm

```diff
@@ -1,104 +1,105 @@
-			/obj/item/gun/ballistic/automatic = "GUN_AUTOMATIC",
-			/obj/item/gun/ballistic/revolver = "GUN_REVOLVER",
-			/obj/item/gun/ballistic/rifle = "GUN_RIFLE",
-			/obj/item/gun/ballistic/shotgun = "GUN_SHOTGUN",
-			/obj/item/stack/sheet = "SHEET",
-			/obj/item/stack/sheet/mineral = "MINERAL_SHEET",
-			/obj/item/stack/ore = "ORE",
-			/obj/item/ai_module = "AI_LAW_MODULE",
-			/obj/item/circuitboard = "CIRCUITBOARD",
-			/obj/item/circuitboard/machine = "MACHINE_BOARD",
-			/obj/item/circuitboard/computer = "COMPUTER_BOARD",
-			/obj/item/reagent_containers = "REAGENT_CONTAINERS",
-			/obj/item/reagent_containers/applicator/pill = "PILL",
-			/obj/item/reagent_containers/applicator/patch = "MEDPATCH",
-			/obj/item/reagent_containers/hypospray/medipen = "MEDIPEN",
-			/obj/item/reagent_containers/cup/glass = "DRINK",
-			/obj/item/food = "FOOD",
-			/obj/item/bodypart = "BODYPART",
-			/obj/effect/decal/cleanable = "CLEANABLE",
-			/obj/item/radio/headset = "HEADSET",
-			/obj/item/clothing = "CLOTHING",
-			/obj/item/clothing/accessory = "ACCESSORY",
-			/obj/item/clothing/mask/gas = "GASMASK",
-			/obj/item/clothing/mask = "MASK",
-			/obj/item/clothing/gloves = "GLOVES",
-			/obj/item/clothing/shoes = "SHOES",
-			/obj/item/clothing/under/plasmaman = "PLASMAMAN_SUIT",
-			/obj/item/clothing/under = "JUMPSUIT",
-			/obj/item/clothing/suit/armor = "ARMOR",
-			/obj/item/clothing/suit = "SUIT",
-			/obj/item/clothing/head/helmet = "HELMET",
-			/obj/item/clothing/head = "HEAD",
-			/obj/item/clothing/neck = "NECK",
-			/obj/item/storage/backpack = "BACKPACK",
-			/obj/item/storage/belt = "BELT",
-			/obj/item/storage/pill_bottle = "PILL_BOTTLE",
-			/obj/item/book/manual = "MANUAL",
-
-			/obj/structure = "STRUCTURE",
-			/obj/structure/closet = "CLOSET",
-			/obj/structure/closet/crate = "CRATE",
-			/obj/structure/closet/crate/secure = "LOCKED_CRATE",
-			/obj/structure/closet/secure_closet = "LOCKED_CLOSET",
-
-			/obj/machinery = "MACHINERY",
-			/obj/machinery/atmospherics = "ATMOS_MECH",
-			/obj/machinery/portable_atmospherics = "PORT_ATMOS",
-			/obj/machinery/door = "DOOR",
-			/obj/machinery/door/airlock = "AIRLOCK",
-			/obj/machinery/rnd/production = "RND_FABRICATOR",
-			/obj/machinery/computer = "COMPUTER",
-			/obj/machinery/computer/camera_advanced/shuttle_docker = "DOCKING_COMPUTER",
-			/obj/machinery/vending = "VENDING",
-			/obj/machinery/vending/wardrobe = "JOBDROBE",
-			/obj/effect = "EFFECT",
-			/obj/projectile = "PROJECTILE",
-		)
 		// ignore_root_path so we can draw the root normally
-		types_to_replacement = zebra_typecacheof(work_from, ignore_root_path = TRUE)
+		types_to_replacement = zebra_typecacheof(GLOB.fancy_type_replacements, ignore_root_path = TRUE)
 		replacement_to_text = list()
-		for(var/key in work_from)
-			replacement_to_text[work_from[key]] = "[key]"
+		for(var/key in GLOB.fancy_type_replacements)
+			replacement_to_text[GLOB.fancy_type_replacements[key]] = "[key]"
```

Comment:
this should really be cached as a local var this is a LOT of lookups (All datums in the game)